include/api_common.php

```diff
@@ -34,15 +55,41 @@ function can_access_project($projectid)
     }
     if (!checkUserPolicy(@$_SESSION['cdash']['loginid'], $projectid, 1)) {
         if ($logged_in) {
-            $response['error'] = 'You do not have permission to access this page.';
-            echo json_encode($response);
-            http_response_code(403);
+            $response = ['error' => 'You do not have permission to access this page.'];
+            json_error_response($response, 403);
         } else {
-            $response['requirelogin'] = 1;
-            echo json_encode($response);
-            http_response_code(401);
+            $response = ['requirelogin' => 1];
+            json_error_response($response, 401);
         }
         return false;
     }
     return true;
 }
+
+/**
+ * Pulls the buildid from the request
+ *
+ * @param bool $required
+ * @return int
+ */
+function get_request_build_id($required = true)
+{
+    $buildid = isset($_REQUEST['buildid']) ? $_REQUEST['buildid'] : null;
+    if ($required && (!$buildid || !is_numeric($buildid))) {
+        json_error_response(['error' => 'Valid build ID required']);
```

Comment:
Could we add a check here that the build exists (`$build->Exists()`) and the user is allowed to see it (`can_access_project($projectid)`)?